### PR TITLE
XFAIL clang/Driver/test/compress.c on AIX 

### PR DESCRIPTION
**Pull request by azhan92**
This PR XFAIL's `clang/Driver/compress.c` on AIX since it requires the LLVM integrated assembler / assembly parser support, which is not available on AIX. 

```
/home/jenkinsWorker/jenkins/workspace/LLVM-Release/release-testing-prototype/llvm/clang18.1.2-final/final/llvm-project/clang/test/Driver/compress.c:19:18: error: CHECK-OPT_GZ: expected string not found in input
// CHECK-OPT_GZ: "--compress-debug-sections=zlib"
```

### CODE REVIEW

File: clang/test/Driver/compress.c [hunk 1]
```diff
@@ -1,5 +1,9 @@
+// XFAIL: system-aix
+
 // REQUIRES: zlib
 
+; XFAIL: target={{.*}}-aix{{.*}}
```
Comment (daltenty) [inline clang/test/Driver/compress.c:5]: You should use the C++ comment format here:
```suggestion
// XFAIL: target={{.*}}-aix{{.*}}
```

**Review by daltenty — APPROVED** [review]
LGTM, thanks!

File: clang/test/Driver/compress.c [hunk 2]
```diff
@@ -1,5 +1,9 @@
+// XFAIL: system-aix
+
 // REQUIRES: zlib
 
+// XFAIL: target={{.*}}-aix{{.*}}
```
Comment (MaskRay) [inline clang/test/Driver/compress.c:5]: system-aix is sufficient and `// XFAIL: target={{.*}}-aix{{.*}}` is uneeded?

Comment (alisonzhang4) [reply clang/test/Driver/compress.c]: That seems correct, I've removed `// XFAIL: target={{.*}}-aix{{.*}}`

**Comment by MaskRay** [comment]
Can you edit the description to state which command fails on AIX?

These tests all contain `-###` to stop after clangDriver completes. And `clang --target=powerpc-ibm-aix -fintegrated-as -Wa,-compress-debug-sections -c a.c '-###'` seems to work for me.

**Comment by azhan92** [comment]
> Can you edit the description to state which command fails on AIX?
> 
> These tests all contain `-###` to stop after clangDriver completes. And `clang --target=powerpc-ibm-aix -fintegrated-as -Wa,-compress-debug-sections -c a.c '-###'` seems to work for me.
> 
> We have other tests that rely on the integrated assembler, so I'd be surprised if this is the only test failing on AIX.

It should be this command that's failing: `RUN: %clang -### -fintegrated-as -gz -c %s 2>&1 | FileCheck -check-prefix CHECK-OPT_GZ %s`. I also added it to the description. The other tests that were failing due to the integrated assembler were XFAIL'd in this [commit](https://github.com/llvm/llvm-project/commit/225cf64e03b4e394c32e95fe1d6414e6e16be094), but it seems like they missed this test since the buildbot doesn't run it (since it requires zlib).

**Comment by MaskRay** [comment]
Thanks for the information. `CHECK-OPT_GZ` is shared by two RUN lines. The failure is due to `-x assembler`.
```
clang '-###' -c --target=powerpc64-ibm-aix -fintegrated-as -gz -x assembler compress.c  # warning: argument unused during compilation: '-gz' [-Wunused-command-line-argument]
```

Since this is unsupported, `UNSUPPORTED:` is more suitable than `XFAIL: system-aix`.

**Review by MaskRay — APPROVED** [review]

**Comment by azhan92** [comment]
@MaskRay, thanks, so would it be better for me to change the XFAIL to UNSUPPORTED in this case?

**Pull request merged**